### PR TITLE
Switch from itsdangerous to pyjwt + FLASK_SECRET fix

### PR DESCRIPTION
**Pull request by QuentiumYT**
Hello,

I'm opening this pull request to submit a new library for generating Json Web Tokens.

As itsdangerous doesn't have JSONWebSignatureSerializer features anymore (https://itsdangerous.palletsprojects.com/en/2.0.x/jws/) I switched to pyjwt.
PyJWT is more specific to this usage and has an active repo as well

I did some testing and everything looks good so far, did some bug fixes as well and changed FLASK_SECRET to SECRET_KEY (I don't know what FLASK_SECRET was refering to, maybe an old flask version)

I also revert your latest commit because of a typo (you can revert it as well before merging this PR if you want to keep it clean)

I'd be happy to see it live because users are constrained using `itsdangerous==2.0.1` with the module

Feel free to ask any change you need

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -45,7 +45,7 @@ something like:
 
   app = flask.Flask(__name__)
   app.config['FLASK_HTPASSWD_PATH'] = '/path/to/.htpasswd'
-  app.config['FLASK_SECRET'] = 'Hey Hey Kids, secure me!'
+  app.config['SECRET_KEY'] = 'Hey Hey Kids, secure me!'
```
Comment (carsongee) [inline README.rst]: Why the change in configuration key? It doesn't seem worth breaking backwards compatibility to me, but maybe there is a good reason. Could you elaborate on why? Thanks!

Comment (QuentiumYT) [reply README.rst]: The reason is that FLASK_SECRET doesn't exist in Flask configs, SECRET_KEY does
I don't know why FLASK_SECRET was there tbh, maybe that was your custom secret name?

Comment (carsongee) [reply README.rst]: Yep, that was intentional so that it wasn't the same key used to sign cookies. It was so long ago though, that I don't fully remember why I found that useful but I would guess that it was due to a concern for having more exposure to crack the key by virtue of having signatures available to a potential attacker in cookies for a given app.

**Comment by carsongee** [comment]
Could you resolve the merge conflicts so that tests can run? Thanks!

File: README.rst (hunk 1 above)
Comment (QuentiumYT) [reply README.rst]: Ohh I see... My bad never thought of that

At first, I thought it was a mistake but kinda unusual to see it everywhere as a bug 😅

Well I can revert it tomorrow if it is planned to keep both keys different.

**Comment by QuentiumYT** [comment]
I did reset your last commit because of a typo to keep the history clean and consistent, that's why I asked if you would like to do so
But yes, I'm perhaps too maniac for this kind of things sorry. Will fix it

**Comment by QuentiumYT** [comment]
All good now :)

**Comment by QuentiumYT** [comment]
I made a clean branch with commit operations in case you prefer merging clean commits
https://github.com/Quentium-Forks/flask-htpasswd/commits/clean
I can open a new PR if you want with this branch

**Comment by QuentiumYT** [comment]
Is there anything needed for this PR? :)

**Comment by QuentiumYT** [comment]
Hello, any update? It's been 1 month and I'm really looking for this fix 😄

**Comment by carsongee** [comment]
Sorry for the delay, and thank you for the contribution!
